samples/charts/data-chart/category-series/Pages/DataChartTypeCategorySeries.razor

```diff
@@ -0,0 +1,342 @@
+                        <option>Spline</option>
+                        <option>SplineArea</option>
+                        <option>StepArea</option>
+                        <option>StepLine</option>
+                        <option>Waterfall</option>
+                    </select>
+
+        <span class="igLegend-title">Legend: </span>
+        <div class="igLegend">
+            <Legend @ref="LegendRef" Orientation="LegendOrientation.Horizontal" />
+        </div>
+    </div>
+
+    <div class="igComponent">
+        @if (DataSource != null)
+        {
+        <DataChart Height="100%" Width="100%"
+                   ChartTitle="Total Olympic Medals"
+                   Subtitle="United States vs China vs Russia"
+                   TitleTopMargin="10"
+                   IsHorizontalZoomEnabled="true"
+                   IsVerticalZoomEnabled="true"
+                   Legend="@LegendRef"
+                   @ref="DataChartRef">
+
+            <CategoryXAxis Name="xAxis" Label="Year" DataSource="@DataSource"
+                           Title="Olympic Years" Gap="0.2" Overlap="-0.05" />
+            <NumericYAxis Name="yAxis" Title="Olympic Medals" TitleLeftMargin="10" />
+
+
+            <NumericXAxis Name="xAxis2" Label="Year" 
+                           Title="Olympic Years" />
+            <CategoryYAxis Name="yAxis2" Title="Olympic Medals" TitleLeftMargin="10" DataSource="@DataSource" Gap="0.2" Overlap="-0.05" />
+
+        </DataChart>
+        }
+        </div>
+</div>
+
+@code {
+
+    private List<SampleCategoryItem> DataSource;
+    private DataChart dataChart;
+    private DataChart DataChartRef
+    {
+        get { return dataChart; }
+        set
+        {
+            dataChart = value;
+            this.setSeries("Column");
+            StateHasChanged();
+        }
+    }
+
+    private Legend legend;
+    private Legend LegendRef
+    {
+        get { return legend; }
+        set
+        {
+            legend = value;
+            StateHasChanged();
+        }
+    }
+    private string seriesType;
+    private string SeriesType
+    {
+        get { return seriesType; }
+        set
+        {
+            seriesType = value;
+            StateHasChanged();
+        }
+    }
+
+    protected override void OnInitialized()
+    {
+        DataChartCoreModule.Register(IgniteUIBlazor);
+        DataChartCategoryCoreModule.Register(IgniteUIBlazor);
+        DataChartCategoryModule.Register(IgniteUIBlazor);
+        DataChartVerticalCategoryModule.Register(IgniteUIBlazor);
+        DataChartInteractivityModule.Register(IgniteUIBlazor);
+
+        LegendModule.Register(IgniteUIBlazor);
+        DataSource = SampleCategoryData.Create();
+    }
+
+    public void OnSeriesTypeChanged(ChangeEventArgs e)
+    {
+        var selectedSeries = e.Value.ToString();
+        SeriesType = selectedSeries;
+        this.setSeries(selectedSeries);
+    }
+
+    public void setSeries(string seriesType)
```

Comment:
there should be no code-behind to create series/axes and all these series/axes must be created in blazor code